itou/siae_evaluations/models.py

```diff
@@ -142,3 +145,146 @@ def __str__(self):
     def clean(self):
         if self.evaluated_period_end_at <= self.evaluated_period_start_at:
             raise ValidationError("La date de début de la période contrôlée doit être antérieure à sa date de fin.")
+
+    def eligible_job_applications(self):
+        # accepted job_applications with self-approval made by hiring siae.
+        return (
+            JobApplication.objects.exclude(approval=None)
+            .select_related("approval", "to_siae", "eligibility_diagnosis", "eligibility_diagnosis__author_siae")
+            .filter(
+                to_siae__department=self.institution.department,
+                to_siae__kind__in=evaluation_enums.EvaluationSiaesKind.Evaluable,
+                state=JobApplicationWorkflow.STATE_ACCEPTED,
+                eligibility_diagnosis__author_kind=KIND_SIAE_STAFF,
+                eligibility_diagnosis__author_siae=F("to_siae"),
+                hiring_start_at__gte=self.evaluated_period_start_at,
+                hiring_start_at__lte=self.evaluated_period_end_at,
+            )
+        )
+
+    def eligible_siaes(self):
+        return (
+            self.eligible_job_applications()
+            .values("to_siae")
+            .annotate(to_siae_count=Count("to_siae"))
+            .filter(to_siae_count__gte=evaluation_enums.EvaluationJobApplicationsBoundariesNumber.MIN)
+        )
+
+    def number_of_siaes_to_select(self):
+        if self.eligible_siaes().count() > 0:
+            return max(round(self.eligible_siaes().count() * self.chosen_percent / 100), 1)
+        return 0
+
+    def eligible_siaes_under_ratio(self):
+        return (
+            self.eligible_siaes().values_list("to_siae", flat=True).order_by("?")[: self.number_of_siaes_to_select()]
+        )
+
+    def eligible_job_applications_under_ratio(self, evaluated_siaes):
+        return [
+            EvaluatedJobApplication(evaluated_siae=evaluated_siae, job_application=job_application)
+            for evaluated_siae in evaluated_siaes
+            for job_application in select_min_max_job_applications(
+                self.eligible_job_applications().filter(to_siae=evaluated_siae.siae)
+            )
+        ]
+
+    def populate_campaign(self):
+
+        if self.evaluations_asked_at:
+
+            return f"La selection de l'échantillon à contrôler a déjà été réalisée pour {self.name}"
```

Comment:
attention, une fonction d'un modèle ne devrait pas retourner un message d'erreur pour l'utilisateur, c'est une erreur de niveau d'abstraction.

on préfèrera:
- raiser en cas d'erreur et ne rien retourner (ce qu'il y a de plus pythonic, better ask for forgiveness than permission)
- retourner un CODE d'erreur (plus C-like)
et dans les deux cas laisser l'appelant (la vue, l'admin) gérer l'affichage intelligible à un utilisateur.

Sinon ta fonction `populate_campaign` sur le modèle perd sa généricité de modèle et se confond avec de l'affichage utilisateur, et se teste différemment.

C'est vrai pour toute fonction en général: traitement et affichage des erreurs != valeurs remontées ou exceptions.